### PR TITLE
fix: Merge upstream 'redo 7.7.23'

### PR DESCRIPTION
**Pull request by richtera**
Redo merge from upstream on 7/7/23 to capture all changes.

- Bump setuptools from 60.9.3 to 65.5.1 in /build_configs/windows
- Bump setuptools from 60.9.3 to 65.5.1 in /build_configs/linux
- Bump setuptools from 60.9.3 to 65.5.1 in /build_configs/macos
- Extract `load_mnemonic_arguments_decorator` func
- [WIP] Add generate_bls_to_execution_change
- Always use `GENESIS_FORK_VERSION` to sign BLSToExecutionChange and fix typo
- Add `0x` prefix to the bytes type fields of the output JSON file
- WIP validation
- Add `--devnet_chain_setting` for devnet testing
- Improve `validate_bls_to_execution_change`
- Add arguments description to README.md
- Echo the folder path
- Arrange JSON fields by adding `metadata`
- Normalize input `bls_withdrawal_credentials` and add `test_existing_mnemonic_bls_withdrawal_interactive`
- Generate multiple changes
- Add checksum address validation
- Add test case of multiple validator indices
- Update README
- Add retype confirmation and fix typo
- Remove shutdown testnets and fill Sepolia and Goerli `genesis_validators_root`
- Add zhejiang, remove Kiln
- Bump version to v2.4.0
- Update README.md
- Apply @james-prysm's suggestions
- Update param description
- Run test_btec_script.py in CI
- Kick the cache
- Run test_binary_btec_script.py in CI
- Strip brackets and extra spaces
- fix error message grammar
- Apply PR feedback by catching the exceptions. Improve error handling and messages.
- Improve error handling
- Confirm deposit `eth1_withdrawal_address` input
- Add more description of mnemonic language per @infosecual's suggestion
- Make `--non_interactive` flag visible and add notes in docs
- clean folders first
- Validate checksum for eth1_withdrawal_address
- Update the checksum tests
- Make `--execution_address` and `--eth1_withdrawal_address` compatible
- Bump version to v2.5.0

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -171,6 +179,22 @@ Success!
...
 ```
 
+###### `generate-bls-to-execution-change` Arguments 
+
+You can use `bls-to-execution-change --help` to see all arguments. Note that if there are missing arguments that the CLI needs, it will ask you for them.
+
+| Argument | Type | Description |
+| -------- | -------- | -------- |
+| `--bls_to_execution_changes_folder` | String. Pointing to `./bls_to_execution_changes` by default | The folder path for the `bls_to_execution_change-*` JSON file(s) |
+| `--chain` | String. `mainnet` by default | The chain setting for the signing domain. |
```
Comment (frozeman) [inline README.md]: This needs to be adopted with the lukso default chain

**Review by frozeman — CHANGES_REQUESTED** [review]
please make sure the readme is correct. see my comment and check if lukso is the default chain

File: README.md (hunk 1 above)
Comment (richtera) [reply README.md]: Checking the diff.

Comment (richtera) [reply README.md]: Added modifications to the --chain table row by copying it from another sub-command.

File: staking_deposit/intl/en/cli/generate_bls_to_execution_change.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
+{
+    "generate_bls_to_execution_change": {
+        "arg_generate_bls_to_execution_change" :{
+            "help": "Generating the SignedBLSToExecutionChange data to enable withdrawals on Ethereum Beacon Chain."
```
Comment (mxmar) [inline staking_deposit/intl/en/cli/generate_bls_to_execution_change.json]: `on Ethereum Beacon Chain` - I'd suggest to change to `on LUKSO Consensus Layer` (Technically it's `beacon chain`).

Comment (mxmar) [reply staking_deposit/intl/en/cli/generate_bls_to_execution_change.json]: Done.

**Review by mxmar — APPROVED** [review]

**Review by mxmar — APPROVED** [review]

**Review by frozeman — APPROVED** [review]

**Pull request merged**